### PR TITLE
KOTOR: Fix room visibility not being evaluated on start

### PR DESCRIPTION
**Pull request by seedhartha**
Also use set for visible rooms instead of a vector.

### CODE REVIEW

File: src/engines/kotorbase/area.cpp [hunk 1]
```diff
@@ -241,13 +241,20 @@ void Area::show() {
 
 	GfxMan.lockFrame();
 
+	std::set<Common::UString> visibleRooms(getRoomsVisibleByPartyLeader());
+
 	// Show rooms
-	for (RoomList::iterator r = _rooms.begin(); r != _rooms.end(); ++r)
-		(*r)->show();
+	for (auto r : _rooms) {
```
Comment (DrMcCoy) [inline src/engines/kotorbase/area.cpp]: should be auto &, and likewise below

Comment (seedhartha) [reply src/engines/kotorbase/area.cpp]: Why though? It would be a reference to a pointer. Makes no sense to me :P

**Comment by seedhartha** [comment]
Fixed foreach loops as requested. I have also restored the logic where doors where always visible (because they may not be recognized as part of the room).

**Comment by DrMcCoy** [comment]
Merged as df7dfd260...6d4fb1313, thanks! :)

**Pull request merged**